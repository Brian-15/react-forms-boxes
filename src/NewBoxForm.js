import React, { useState } from 'react';
import './NewBoxForm.css';

const NewBoxForm = ({ addBox }) => {
    const initialData = { backgroundColor: '', width: '', height: ''};
    const [formData, setFormData] = useState(initialData);

    const handleChange = evt => {
        const { name, value } = evt.target;
        setFormData(data => ({
            ...data,
            [name]: value
        }));
    };
    const handleSubmit = evt => {
        evt.preventDefault();
        if (Object.values(formData).some(val => val === '')) return;
        addBox(formData);
        setFormData(initialData);
    };

    return (
        <form className='NewBoxForm' onSubmit={handleSubmit} data-testid="form">
            <label htmlFor='backgroundColor'>Color:
              <input
                name='backgroundColor'
                value={formData.backgroundColor}
                onChange={handleChange}
                data-testid='bg-color'
              />
            </label>
            <label htmlFor='width'>Width:
              <input
                name='width'
                value={formData.width}
                onChange={handleChange}
                data-testid='width'
              />
            </label>
            <label htmlFor='height'>Height:
              <input
                name='height'
                value={formData.height}
                onChange={handleChange}
                data-testid='height'
              />
            </label>
            <button>Add Box</button>
        </form>
    );
};

export default NewBoxForm;