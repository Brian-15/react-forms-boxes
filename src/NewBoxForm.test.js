import { render } from '@testing-library/react';
import NewBoxForm from './NewBoxForm';

test('smoke test: renders form', () => {
    render(<NewBoxForm />);
});

test('matches snapshot', () => {
    const { asFragment } = render(<NewBoxForm />);
    expect(asFragment()).toMatchSnapshot();
});

