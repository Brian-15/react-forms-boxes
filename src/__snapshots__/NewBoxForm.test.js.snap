// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewBoxForm"
    data-testid="form"
  >
    <label
      for="backgroundColor"
    >
      Color:
      <input
        data-testid="bg-color"
        name="backgroundColor"
        value=""
      />
    </label>
    <label
      for="width"
    >
      Width:
      <input
        data-testid="width"
        name="width"
        value=""
      />
    </label>
    <label
      for="height"
    >
      Height:
      <input
        data-testid="height"
        name="height"
        value=""
      />
    </label>
    <button>
      Add Box
    </button>
  </form>
</DocumentFragment>
`;
